packages/tools/src/EditorFrame.tsx

```diff
@@ -358,6 +327,135 @@ export const EditorFrame = (props: EditorFrameProps) => {
+    const blobInfo = {
+      filename: () => file.name,
+      blob: () => file,
+      blobUri: () => '',
+      base64: reader ? (reader.result as string).split(',')[1] : '',
+    };
+    return blobInfo;
+  };
+
+  const handlePaste = (plugin: unknown, e: PastePreProcessEvent) => {
+    if (e.internal) {
+      return;
+    }
+
+    // Capture the original content
+    let originalContent = e.content;
+
+    // Clear the content to prevent the default paste action
+    e.content = '';
+
+    // Clean the content
+    const isInlineSnippet =
+      !/<(?:p|div|ul|ol|li|table|tbody|thead|caption|tfoot|tr|td|th|blockquote|section|h[1-6])[ >]/i.test(
+        originalContent,
+      );
+    originalContent = dirtyClean(originalContent as HTMLText);
+    if (isInlineSnippet) {
+      originalContent = originalContent
+        .replace(/^<p>/, '')
+        .replace(/<\/p>$/, '');
+    } else {
+      let content = originalContent;
+      if (!/article__title/.test(content)) {
+        content = content.replace(
+          /<p>(\d+\.\s+gr\.(\s+)?)(<br\s+?\/>)?/g,
+          '<h3 class="article__title">$1</h3><p>',
+        );
+      }
+      originalContent = content;
+    }
+
+    if (originalContent.includes('data:image/')) {
+      const imgTagRegex =
+        /<img.*?src=["'](data:image\/(png|jpeg|svg\+xml);base64,.*?)["'].*?>/g;
+      let match;
+      const images: Array<{
+        base64: string;
+        match: string;
+        placeholder: string;
+      }> = [];
+      let contentWithPlaceholders = originalContent;
+
+      // Collect all base64 images from the content
+      while ((match = imgTagRegex.exec(originalContent)) !== null && match[1]) {
+        const placeholder = `__IMAGE_PLACEHOLDER_${images.length}__`;
+        images.push({ base64: match[1], match: match[0], placeholder });
+        contentWithPlaceholders = contentWithPlaceholders.replace(
+          match[0],
+          placeholder,
+        );
+      }
+
+      if (images.length > 0) {
+        let imagesProcessed = 0;
+        images.forEach(({ base64, placeholder }, index) => {
+          const { file, filename } = base64ToBlob(base64 ?? '');
+          if (file) {
+            const blobInfo = convertFileToBlobInfo(file);
+            props.fileUploader(
+              blobInfo,
+              (url: string) => {
+                // Replace the placeholder with the uploaded image URL
+                contentWithPlaceholders = contentWithPlaceholders.replace(
+                  placeholder,
+                  `<img src="${url}" alt="${filename}" />`,
+                );
+
+                imagesProcessed++;
+                // Insert the content once all images are processed
+                if (imagesProcessed === images.length) {
+                  tinymce.activeEditor.insertContent(contentWithPlaceholders);
+                }
+              },
+              (error: string) => {
+                console.error('Image upload failed:', error);
```

Comment:
What happens in the editor UI if this happens?

Reply:
I THINK, it will just paste the text, and leave the image out. I will try.

Reply:
Well it did not, but does now. 🤝